python/foglamp/services/core/api/task.py

```diff
@@ -0,0 +1,244 @@
+        "schedule_day": 0,
+        "schedule_time": 0,
+        "schedule_repeat": 30,
+        "schedule_enabled": true,
+        "cmd_params": {
+            "stream_id": "1",
+            "debug_level": "1"
+        }
+     }'
+    """
+
+    try:
+        data = await request.json()
+        if not isinstance(data, dict):
+            raise ValueError('Data payload must be a dictionary')
+
+        name = data.get('name', None)
+        plugin = data.get('plugin', None)
+        task_type = data.get('type', None)
+
+        schedule_type = data.get('schedule_type', None)
+        schedule_day = data.get('schedule_day', None)
+        schedule_time = data.get('schedule_time', None)
+        schedule_repeat = data.get('schedule_repeat', None)
+        enabled = data.get('schedule_enabled', None)
+
+        cmd_params = data.get('cmd_params', None)
+
+        if name is None:
+            raise web.HTTPBadRequest(reason='Missing name property in payload.')
+        if plugin is None:
+            raise web.HTTPBadRequest(reason='Missing plugin property in payload.')
+        if task_type is None:
+            raise web.HTTPBadRequest(reason='Missing type property in payload.')
+        if utils.check_reserved(name) is False:
+            raise web.HTTPBadRequest(reason='Invalid name property in payload.')
+        if utils.check_reserved(plugin) is False:
+            raise web.HTTPBadRequest(reason='Invalid plugin property in payload.')
+        if task_type not in ['south', 'north']:
+            raise web.HTTPBadRequest(reason='Only north and south types are supported.')
+        if cmd_params is not None:
+            if not isinstance(cmd_params, dict):
+                raise web.HTTPBadRequest(reason='cmd_params must be a dict.')
+
+        if schedule_type is None:
+            raise web.HTTPBadRequest(reason='Schedule Type is mandatory')
+        if not isinstance(schedule_type, int) and not schedule_type.isdigit():
+            raise web.HTTPBadRequest(reason='Error in type: {}'.format(schedule_type))
+        if int(schedule_type) not in list(Schedule.Type):
+            raise web.HTTPBadRequest(reason='Schedule type error: {}'.format(schedule_type))
+        if int(schedule_type) == Schedule.Type.STARTUP:
+            raise web.HTTPBadRequest(reason='Schedule type cannot be STARTUP: {}'.format(schedule_type))
+
+        schedule_type = int(schedule_type)
+
+        if schedule_day is not None:
+            if isinstance(schedule_day, float) or (isinstance(schedule_day, str) and (schedule_day.strip() != "" and not schedule_day.isdigit())):
+                raise web.HTTPBadRequest(reason='Error in day: {}'.format(schedule_day))
+        else:
+            schedule_day = int(schedule_day) if schedule_day is not None else None
+
+        if schedule_time is not None and (not isinstance(schedule_time, int) and not schedule_time.isdigit()):
+            raise web.HTTPBadRequest(reason='Error in time: {}'.format(schedule_time))
```

Comment:
return  the error with required format info and actual payload property name i.e. schedule_time